### ISSUE TITLE
error and logged events need more context

### ISSUE THREAD

**Issue by pekeler** [getsentry/raven-node#215]
When my error and logged event handlers gets called, I want to proceed or exit depending on context. Specifically, I want to know whether this is from a message or an exception. Ideally, the event includes the actual message/exception I was trying to send. At least something like `isMessage`.

``` JavaScript
ravenClient.on('logged', function(info) {
  if (!info.isMessage) {
    setImmediate(() => process.exit(1))
  }
})
ravenClient.on('error', function(error) {
  console.error(error.reason, error.statusCode)
  if (!error.isMessage) {
    process.exit(1)
  }
})
```

Is this something that can be added?

**Comment by kamilogorek** [comment]
I'm not sure if I understand what's the context and what you want to achieve. Could you please provide more information?

**Comment by pekeler** [comment]
When raven detected an error and successfully logged it to Sentry, the `logged` event handler gets called.

When my code called `catureMessage()` and it was successfully logged to Sentry, the very same `logged` event handler gets called.

In the former case, I want to kill my process. In the latter, I want to do nothing. The eventhandler doesn't provide me with enough information to make that decision.

The `error` event handler has the same limitation.

---

What I really want is reliable way to kill my process after an error, but only after the error was successfully logged to Sentry.

**Comment by kamilogorek** [comment]
We do have callbacks for `captureMessage` and `captureException`, which can allow you to make a distinction which events has been triggered manually, but not other way around.

What do you think about it @benvinegar?

The problem with this solution is that we'd have to pass this information from the very first call.

`process's uncaughtException  => makeErrorHandler/errorHandler => captureException => process => send => self.transport.send => client.emit('logged')`

**Comment by kamilogorek** [comment]
Since this issue would require a large rewrite, I'll close it for now. We may tackle it when working on v3. Feel free to open a new issue/reopen if it's still relevant.
